packages/js/src/ui/helpers/useUncontrolledState.ts

```diff
@@ -0,0 +1,33 @@
+import { Accessor, createSignal } from 'solid-js';
+
+type UseUncontrolledState = {
+  value?: boolean;
+  fallbackValue?: boolean;
+};
+
+type UseUncontrolledStateOutput = [Accessor<boolean>, (value: boolean) => void];
+
+export function useUncontrolledState({
+  value,
+
+  fallbackValue,
+}: UseUncontrolledState): UseUncontrolledStateOutput {
+  const [uncontrolledValue, setUncontrolledValue] = createSignal(!!fallbackValue);
+
+  const handleUncontrolledChange = (val: boolean) => {
+    setUncontrolledValue(val);
+  };
+  /**
+   * If value is provided, return controlled state
+   */
+  if (value !== undefined) {
+    const accessor: Accessor<boolean> = () => value;
+
+    return [accessor, handleUncontrolledChange];
+  }
+
+  /**
+   * If value is not provided, return uncontrolled state
+   */
+  return [uncontrolledValue, handleUncontrolledChange];
```

Comment:
Is returning accessor a pattern with solid? cc: @SokratisVidros

Reply:
@desiprisg, To have reactivity, we need to pass around accessors
Here is a small snippet to rest reactivity with and without accessors
https://stackblitz.com/edit/github-9nx6bn?file=src%2FApp.tsx

Reply:
Ah yes this acts like a `createSignal` itself, right.